### DISCUSSION TITLE
How can i upload a folder

### DISCUSSION THREAD

**Discussion by tanyaofei** [fastapi/fastapi#7545] [Questions] [answered]
How can i upload a folder

**Comment by tiangolo** [comment]
I assume you solved your issue, so thanks for reporting back and closing the issue :+1:

**Reply by HolzmanoLagrene** [reply]
I know its like 4 years later, but how do you upload a folder? I know there are ways. Any way to do it using fastapi?

**Comment by YuriiMotov** [comment, marked as answer]
I think it can be done in two (at least) ways:
* on client side create a zip file that folder and upload it as one file. Decompress it on server side
* on client side send all files with their relative paths. On server side handle multiple files upload and reconstruct the folder using relative paths
